fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -545,13 +617,16 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 	UID id;
 	Reference<IThreadPool> writeThread;
 	Reference<IThreadPool> readThreads;
-	Promise<Void> errorPromise;
+	std::shared_ptr<RocksDBErrorListener> errorListener;
```

Comment:
Is this needed? I don't _think_ the lifetime of the `Future` is attached to the lifetime of the `ThreadReturnPromise`, so it is fine to have the `RocksDBErrorListener` only owned by the RocksDB instance.

Reply:
Right, it's not needed.

Reply:
I was actually trying to push in the other direction, e.g. removing this and leaving the `Future`. This method, while not particularly expensive, requires grabbing a lock among other things, while a copy is just incrementing a counter.

Reply:
I think we still need errorListener because getError() could be called before init().  Blocking the main thread is probably not a good idea. I made the change. That brings another question, Is incrementing the [count](https://github.com/apple/foundationdb/blob/e68b131e4a23d7a780cc3767fc6bd1cd31a21966/flow/flow.h#L715) thread safe?

Reply:
Ah, now I understand. Thanks! That's quite unfortunate.

Incrementing the `count` is not thread safe. That's why `getFuture()` can only be called from the main thread.